config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -424,6 +431,30 @@ private String generateBundleFileName(boolean isDeleteBundle, String bundleName)
         return bundleName + filenameSuffix;
     }
 
+    public void addPrivateKeyContexts(Bundle bundle, ProjectInfo projectInfo, BundleArtifacts bundleArtifacts,
+                                      Document document) {
+        if (!bundle.getPrivateKeys().isEmpty()) {
+            bundle.getPrivateKeys().forEach((alias, privateKey) -> {
+                if (privateKey.getPrivateKeyFile() == null) {
+                    privateKey.setPrivateKeyFile(bundle.getPrivateKeyFiles().get(privateKey.getAlias()));
+                }
+                if (privateKey.getPrivateKeyFile() != null) {
+                    Element element = privateKeyImportContextBuilder.build(privateKey, document);
+                    String filename = generatePrivateKeyFileName(privateKey, projectInfo);
+                    bundleArtifacts.addPrivateKeyContext(element, filename);
+                }
+            });
+        }
+    }
+
+    private String generatePrivateKeyFileName(PrivateKey privateKey, ProjectInfo projectInfo) {
+        StringBuilder filename = new StringBuilder(projectInfo.getName()).append("-").append(privateKey.getAlias());
+        if (StringUtils.isNotBlank(projectInfo.getVersion())) {
+            filename.append("-").append(projectInfo.getVersion());
```

Comment:
is config name not required in the file name ?

Reply:
@uppoju We can add it. @graju256 should I make naming exactly the same as Environment bundle naming?

Reply:
Nope. I could see the benefit in distinguishing them from the bundle from its extension. Probably, we can remove the .xml from the extension, so that it would become more opt to the sitation.

{name}-{version}.privatekey

@uppoju what do you think?

Reply:
@graju256, can this be used for boot strapping, if yes whether file extension should be always xml ?

Reply:
if not, we can have privatekey extenstion

Reply:
Nope, this doesn't suite for bootstrapping.

Reply:
Added configName as well in the filename and removed `.xml`.